src/components/ConfirmLeave.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+/**
+ * Ask the user for confirmation to leave the page.
+ * @extends React.Component
+ *
+ * @param props {object}
+ * @param props.when {boolean} - Whether the component should block
+ * @param props.message {string} - The message the browser shows, maybe
+ */
+class ConfirmLeave extends React.Component {
+  static defaultProps = {
+    message: 'Are you sure you want to leave the page with unsaved changes?',
```

Comment:
Since they can't actually save any changes, maybe something like, 'This action will erase all current data. Are you sure you want to do this?' would be more appropriate. Not a blocking note, but one I want to mark down for a next step.

Reply:
Sure. I'll change the other instances as well.

Reply:
Maybe this belongs in VisitPage.js.